src/foam/lib/parse/AbstractLiteral.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package foam.lib.parse;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public abstract class AbstractLiteral
+  implements Parser
+{
+  protected final static Map map__ = new ConcurrentHashMap();
```

Comment:
This is shared across all Parsers - not an issue?  I would expect to create ThreadLocal parsers.

Reply:
I think not an issue, we are creating singleton for Literal's that only specify a character without specific value for that character